### PR TITLE
DTSPO-14865-replace-kured-v2

### PR DESCRIPTION
**Pull request by JoshuaGlassKainos**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-14865

### Change description ###
Added maintenance window config obj

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: components/aks/aks.tf [hunk 1]
```diff
@@ -138,6 +138,14 @@ module "kubernetes" {
   enable_automatic_channel_upgrade_patch = var.enable_automatic_channel_upgrade_patch
 
   enable_node_os_channel_upgrade_nodeimage = contains(["sbox"], var.env) ? true : false
+
+  node_os_maintenance_window_duration    = var.node_os_maintenance_window_duration
+  node_os_maintenance_window_frequency   = var.node_os_maintenance_window_frequency
+  node_os_maintenance_window_day_of_week = var.node_os_maintenance_window_day_of_week
```
Comment (JoshuaGlassKainos) [inline components/aks/aks.tf:144]: not sure if we want to specify a weekly freq. just restrict it to daily only

**Comment by JoshuaGlassKainos** [comment]
will re run test once module has been updated

File: components/aks/aks.tf [hunk 2]
```diff
@@ -138,6 +138,16 @@ module "kubernetes" {
   enable_automatic_channel_upgrade_patch = var.enable_automatic_channel_upgrade_patch
 
   enable_node_os_channel_upgrade_nodeimage = contains(["sbox"], var.env) ? true : false
+
+  node_os_maintenance_window_config = {
```
Comment (MartyFox) [inline components/aks/aks.tf:142]: If this code is in the module, do we need it here?
Should this be: `node_os_maintenance_window_config = var. node_os_maintenance_window_config`?

File: environments/aks/sbox.tfvars [hunk 3]
```diff
@@ -40,3 +40,9 @@ windows_node_pool = {
 availability_zones = ["1"]
 
 autoShutdown = true
+
+node_os_maintenance_window_config = {
+  frequency   = "Daily"
+  duration    = 4
```
Comment (MartyFox) [inline environments/aks/sbox.tfvars:46]: Can remove the 4 as its default, could increase to 5 if you are testing to be sure its picking it up.

File: components/aks/inputs-default.tf [hunk 4]
```diff
@@ -65,3 +65,16 @@ variable "kube_audit_admin_logs_enabled" {
...
   default = false
 }
+
+variable "node_os_maintenance_window_config" {
+  type = object({
+    frequency   = optional(string, "Weekly")
+    interval    = optional(number, 1)
+    duration    = optional(number, 4)
+    day_of_week = optional(string, "Monday")
+    start_time  = optional(string, "18:00")
```
Comment (MartyFox) [inline components/aks/inputs-default.tf:75]: Given teams do out of hours releases do we want a maintenance window starting at 6pm?
At the same time we turn off most clusters at night so this is a difficult time to figure out.

Comment (JoshuaGlassKainos) [reply components/aks/inputs-default.tf:75]: the timing of the window would need further discussion, to see what would best for a default value for the reason you mentioned.

File: components/aks/aks.tf (hunk 2 above)
Comment (JoshuaGlassKainos) [reply components/aks/aks.tf:142]: good shout, will test to see if this is needed or not.

**Pull request merged**